### ISSUE TITLE
respawn tru inspector and value user override

### ISSUE THREAD

**Issue by Ardescar** [Frooxius/NeosPublic#629]
By inspecting the items of a valueuseroverride the avatar gets removed and the user does respawn.
See attached video.

[respawntroughinspector.zip] 
  10 mb (https://github.com/Frooxius/NeosPublic/files/4713434/respawntroughinspector.zip)

**Comment by shiftyscales** [comment]
[SHIFTY-PC - 2020.5.28.1239 - 2020-06-01 14_20_11.log](https://github.com/Frooxius/NeosPublic/files/4713506/SHIFTY-PC.-.2020.5.28.1239.-.2020-06-01.14_20_11.log)
```
2:22:33 PM.400 ( 58 FPS)	Exception in updating Tool on Element: IDADD00, Type: FrooxEngine.ToolSlot, World: Shifty World, IsRemoved: False, IsDestroyed: False, Enabled: True
Element: IDA6A00, Type: FrooxEngine.Slot, World: Shifty World, IsRemoved: False, Slot name: RightController, T: [0.9385664; 0.9485303; -1.223505], R: [-0.3254267; 0.3764359; -0.1325309; 0.8572217], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID4B400, Type: FrooxEngine.Slot, World: Shifty World, IsRemoved: False, Slot name: User Shifty (ID2900), T: [1.159416; -0.02467048; -0.41917], R: [0; 0.2898518; 0; 0.9570715], S: [1.371429; 1.371429; 1.371429], ActiveSelf: True, IsDestroyed: False
Element: ID1E00, Type: FrooxEngine.Slot, World: Shifty World, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: Shifty World, IsRemoved: False

System.ArgumentOutOfRangeException: Length cannot be less than zero.
Parameter name: length
  at System.String.Substring (System.Int32 startIndex, System.Int32 length) [0x0004a] in <437ba245d8404784b9fbab9b439ac908>:0 
  at BaseX.ReflectionExtensions.GetNiceName (System.Type type, System.String open, System.String close) [0x00053] in <f4baf1bbb54b4b73a0a5c9be86c0bd92>:0 
  at FrooxEngine.WorkerInspector.BuildUIForComponent (FrooxEngine.Worker worker, System.Boolean allowRemove, System.Boolean allowDuplicate, System.Predicate`1[T] memberFilter) [0x0006f] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.WorkerInspector.Setup (FrooxEngine.Worker target, System.Predicate`1[T] memberFilter) [0x0000c] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.WorkerInspector.Create (FrooxEngine.Slot root, FrooxEngine.Worker worker, System.Predicate`1[T] memberFilter) [0x00113] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.InspectorHelper.OpenInspectorForTarget (FrooxEngine.IWorldElement target, FrooxEngine.Slot source, System.Boolean openWorkerOnly) [0x00122] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.ReferenceProxy.Trigger () [0x00029] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.CommonTool.OnActionStart (FrooxEngine.BasicAction action) [0x0016e] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.Tool.UpdateTool (FrooxEngine.IStandardController input) [0x00017] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.CommonTool.UpdateTool (FrooxEngine.IStandardController input) [0x003df] in <86556071e05b4c7b840fdd89fe339c79>:0 
  at FrooxEngine.ToolSlot.OnCommonUpdate () [0x00119] in <86556071e05b4c7b840fdd89fe339c79>:0 

   at void BaseX.UniLog.Error(string message, bool stackTrace)
   at void FrooxEngine.DebugManager.Error(object obj, bool stackTrace)
   at void FrooxEngine.ToolSlot.OnCommonUpdate()
   at void FrooxEngine.ComponentBase<FrooxEngine.Component>.InternalRunUpdate()
   at bool FrooxEngine.UpdateManager.RunUpdates()
   at bool FrooxEngine.World.RefreshStep()
   at bool FrooxEngine.World.Refresh()
   at void FrooxEngine.WorldManager.UpdateStep(double maxMilliseconds)
   at bool FrooxEngine.WorldManager.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngine.Engine.UpdateStep(double maxMilliseconds)
   at void FrooxEngine.Engine.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngineRunner.Update()
2:22:34 PM.926 ( 60 FPS)	Respawning user: Shifty (ID2900)
```

**Comment by Frooxius** [comment]
What is the datatype used for the ValueUserOverride?

**Comment by shiftyscales** [comment]
Integer as depicted in his video.

**Comment by shiftyscales** [comment]
It appears to occur with any arbitary type too, though. Just grip on "Item" and press trigger to open the override inspector which will respawn the user.

**Comment by Frooxius** [comment]
Fixed in 2020.6.2.95, thanks!
